src/main/java/gregtech/api/enums/VoidingMode.java

```diff
@@ -54,6 +70,24 @@ public VoidingMode previous() {
         return values()[(ordinal() + values().length - 1) % values().length];
     }
 
+    public VoidingMode nextInCollection(Collection<VoidingMode> allowed) {
+        if (allowed.isEmpty()) throw new IllegalArgumentException("nothing allowed");
+        VoidingMode ret;
+        do {
+            ret = next();
```

Comment:
```suggestion
        VoidingMode ret = this;
        do {
            ret = ret.next();
```
Otherwise it results in infinite loop